testing/node-driver/src/main/kotlin/net/corda/testing/internal/DriverDSLImpl.kt

```
@@ -96,11 +98,25 @@ class DriverDSLImpl(
 
     //TODO: remove this once we can bundle quasar properly.
     private val quasarJarPath: String by lazy {
-        val cl = ClassLoader.getSystemClassLoader()
-        val urls = (cl as URLClassLoader).urLs
-        val quasarPattern = ".*quasar.*\\.jar$".toRegex()
-        val quasarFileUrl = urls.first { quasarPattern.matches(it.path) }
-        Paths.get(quasarFileUrl.toURI()).toString()
+        resolveJar(".*quasar.*\\.jar$")
+    }
+
+    private val jolokiaJarPath: String by lazy {
+        resolveJar(".*jolokia-jvm-.*-agent\\.jar$")
+    }
+
+    private fun resolveJar(jarNamePattern: String): String {
+        return try {
+            val cl = ClassLoader.getSystemClassLoader()
+            val urls = (cl as URLClassLoader).urLs
+            val jarPattern = jarNamePattern.toRegex()
+            val jarFileUrl = urls.first { jarPattern.matches(it.path) }
+            Paths.get(jarFileUrl.toURI()).toString()
+        }
+        catch(e: Exception) {
+            log.warn("Unable to locate JAR `$jarNamePattern` on classpath: ${e.message}")
```

If we log, `log.warn(message, e)` is better. Maybe for the driver is better to throw instead.

---

Maybe not only for the driver, if the configuration specifies to start the JXM HTTP server, and no jolokia agent is in the classpath, it's better to fail at startup, rather than log and ignore.

---

As discussed in latest stand-up, I will make the node fail-fast if the jolokia-agent-jvm.jar file is not located.